styling errors on Tomcat after upgrade to 5.4.2 when packaging for prod

updated it after bellow investigation:

##### **Overview of the issue**

After doing an upgrade from 5.3.1 to 5.4.2 all seemed to work fine when working from my developer machine, but after doing a package and deploy to tomcat the fonts are wrong and i noticed that the main.[hash].css is empty and in a different location. 
there were also strange chars comming from javascript files like nbsp; had an A in front and euro sign was strange.

firefox can not find the man.*.css file which is empty

Besides moving from 5.3.1 to 5.4.2 i also switched from yarn to npm.

i have found end of pipeline solutions to my problems but did not find root cause.

1) the fonts were back when i added **!important**, this suggests the order is different than before i did the upgrade
2) i needed to add  **charset="UTF-8"** to the script import tags in the index.html files. i do not know why this is needed now and not in the previous version. 
Adding this by hand is not good is there a way to add this automatically or make it not needed any more?

##### **JHipster Version(s)**

5.4.2

##### **JHipster configuration**
 jhipster info
Using JHipster version installed locally in current project's node_modules
Executing jhipster:info
Options: 
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
stand-bystand@0.0.0 /home/tibi/git/standBystand
└── (empty)

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
</pre>
</details>


##### **Environment and Tools**

openjdk version "1.8.0_181"
OpenJDK Runtime Environment (build 1.8.0_181-8u181-b13-0ubuntu0.18.04.1-b13)
OpenJDK 64-Bit Server VM (build 25.181-b13, mixed mode)

git version 2.17.1

node: v8.11.3

npm: 6.2.0

yeoman: 2.0.5

yarn: 1.9.4

.yo-rc.json:


{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "nl.tibi.sbys",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "5.4.2",
    "baseName": "standBystand",
    "packageName": "nl.tibi.sbys",
    "packageFolder": "nl/tibi/sbys",
    "serverPort": "8080",
    "authenticationType": "session",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "mysql",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSocialSignIn": true,
    "enableSwaggerCodegen": false,
    "rememberMeKey": "xxxxxxxxxx",
    "clientFramework": "angularX",
    "useSass": false,
    "clientPackageManager": "npm",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "nl",
      "de"
    ]
  }
}

##### **Browsers and Operating System**

ubuntu 18.04

-   [x] Checking this box is mandatory (this is just to show you read everything)

```
tibi@tibi:~/git/tmpJhipsterTest/test-app$ node ../generator-jhipster-fix/cli/cli.js
module.js:549
    throw err;
    ^

Error: Cannot find module 'commander'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/tibi/git/tmpJhipsterTest/generator-jhipster-fix/cli/cli.js:20:17)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)

```

--- a/generators/client/templates/angular/webpack/webpack.prod.js.ejs
+++ b/generators/client/templates/angular/webpack/webpack.prod.js.ejs
@@ -49,7 +49,7 @@ module.exports = webpackMerge(commonConfig({ env: ENV }), {
         main: './<%= MAIN_SRC_DIR %>app/app.main'
     },
     output: {
-        path: utils.root('<%= BUILD_DIR %><% if (buildTool === 'maven') { %>classes/public<% } %><% if (buildTool === 'gradle'){ %>resources/main/public<% } %>'),
+        path: utils.root('<%= BUILD_DIR %>www'),
         filename: 'app/[name].[hash].bundle.js',
         chunkFilename: 'app/[id].[hash].chunk.js'
     },

--- a/generators/client/templates/react/webpack/webpack.prod.js.ejs
+++ b/generators/client/templates/react/webpack/webpack.prod.js.ejs
@@ -40,7 +40,7 @@ module.exports = webpackMerge(commonConfig({ env: ENV }), {
     main: './<%= MAIN_SRC_DIR %>app/index'
   },
   output: {
-    path: utils.root('<%= BUILD_DIR %><% if (buildTool === 'maven') { %>classes/public<% } %><% if (buildTool === 'gradle'){ %>resources/main/public<% } %>'),
+    path: utils.root('<%= BUILD_DIR %>www'),
     filename: 'app/[name].[hash].bundle.js',
     chunkFilename: 'app/[name].[hash].chunk.js'
   },

--- a/generators/common/templates/README.md.ejs
+++ b/generators/common/templates/README.md.ejs
@@ -292,7 +292,7 @@ To stop it and remove the container, run:
 You can also fully dockerize your application and all the services that it depends on.
 To achieve this, first build a docker image of your app by running:
 
-    <% if (buildTool === 'maven') { %>./mvnw package -Pprod jib:dockerBuild<% } %><% if (buildTool === 'gradle') { %>./gradlew bootWar -Pprod jibDockerBuild<% } %>
+    <% if (buildTool === 'maven') { %>./mvnw package -Pprod verify jib:dockerBuild<% } %><% if (buildTool === 'gradle') { %>./gradlew bootWar -Pprod jibDockerBuild<% } %>
 
 Then run:
 

--- a/generators/docker-base.js
+++ b/generators/docker-base.js
@@ -46,7 +46,7 @@ function checkImages() {
         const appConfig = this.appConfigs[index];
         if (appConfig.buildTool === 'maven') {
             imagePath = this.destinationPath(`${this.directoryPath + appsFolder}/target/jib-cache`);
-            runCommand = './mvnw package -Pprod jib:dockerBuild';
+            runCommand = './mvnw package -Pprod verify jib:dockerBuild';
         } else {
             imagePath = this.destinationPath(`${this.directoryPath + appsFolder}/build/jib-cache`);
             runCommand = './gradlew bootWar -Pprod jibDockerBuild';

--- a/generators/docker-utils.js
+++ b/generators/docker-utils.js
@@ -85,7 +85,7 @@ function checkImageExist(opts = { cwd: './', appConfig: null }) {
     this.warningMessage = 'To generate the missing Docker image(s), please run:\n';
     if (opts.appConfig.buildTool === 'maven') {
         imagePath = this.destinationPath(`${opts.cwd + opts.cwd}/target/docker`);
-        this.dockerBuildCommand = './mvnw package -Pprod jib:dockerBuild';
+        this.dockerBuildCommand = './mvnw package -Pprod verify jib:dockerBuild';
     } else {
         imagePath = this.destinationPath(`${opts.cwd + opts.cwd}/build/docker`);
         this.dockerBuildCommand = './gradlew bootWar -Pprod jibDockerBuild';

--- a/generators/server/templates/gradle/docker.gradle.ejs
+++ b/generators/server/templates/gradle/docker.gradle.ejs
@@ -27,11 +27,6 @@ buildscript {
 
 apply plugin: com.google.cloud.tools.jib.gradle.JibPlugin
 
-task copyWwwIntoStatic (type: Copy) {
-    from 'build/www/'
-    into 'build/resources/main/static'
-}
-
 jib {
     from {
         image = '<%= DOCKER_JAVA_JRE %>'
@@ -51,4 +46,11 @@ jib {
     }
 }
 
+<%_ if (!skipClient) { _%>
+task copyWwwIntoStatic (type: Copy) {
+    from '<%= CLIENT_DIST_DIR %>'
+    into 'build/resources/main/static'
+}
+
 jibDockerBuild.dependsOn copyWwwIntoStatic
+<%_ } _%>

--- a/generators/server/templates/gradle/docker.gradle.ejs
+++ b/generators/server/templates/gradle/docker.gradle.ejs
@@ -27,6 +27,11 @@ buildscript {
 
 apply plugin: com.google.cloud.tools.jib.gradle.JibPlugin
 
+task copyWwwIntoStatic (type: Copy) {
+    from 'build/www/'
+    into 'build/resources/main/static'
+}
+
 jib {
     from {
         image = '<%= DOCKER_JAVA_JRE %>'
@@ -45,3 +50,5 @@ jib {
         useCurrentTimestamp = true
     }
 }
+
+jibDockerBuild.dependsOn copyWwwIntoStatic

--- a/generators/server/templates/pom.xml.ejs
+++ b/generators/server/templates/pom.xml.ejs
@@ -1075,13 +1075,13 @@
                             </configuration>
                         </execution>
                         <execution>
-                            <id>docker-resources</id>
+                            <id>jib-www-resources</id>
                             <phase>verify</phase>
                             <goals>
                                 <goal>copy-resources</goal>
                             </goals>
                             <configuration>
-                                <outputDirectory>target/classes/static/</outputDirectory>
+                                <outputDirectory>target/classes/public/</outputDirectory>
                                 <resources>
                                     <resource>
                                         <directory>target/www</directory>

--- a/generators/server/templates/pom.xml.ejs
+++ b/generators/server/templates/pom.xml.ejs
@@ -1081,7 +1081,7 @@
                                 <goal>copy-resources</goal>
                             </goals>
                             <configuration>
-                                <outputDirectory>target/classes/public/</outputDirectory>
+                                <outputDirectory>target/classes/static/</outputDirectory>
                                 <resources>
                                     <resource>
                                         <directory>target/www</directory>

